### DISCUSSION TITLE
Custom templates on Arch Linux package

### DISCUSSION THREAD

**Discussion by scossu** [gogs/gogs#7059] [Q&A]
Hi,

I installed Gogs from an Archlinux package: https://aur.archlinux.org/packages/gogs and now I'd like to customize its default theme and logo.

The [templating documentation](https://gogs.io/docs/features/custom_template) specifies the `templates/` and `custom/templates/` directories as the locations to look for, but I cannot find those in the list of files provided by the Archlinux package. Furthermore, I cannot find any of the CSS and image assets that are served by the Gogs web server in the package files. 

Is it possible to modify a template in this kind of setup? Or do I need to work off the source code and deploy the application myself? 

Note: I have no knowledge of how Go source code and assets are compiled and packaged, so consider this a "non-Go dev" use-case... If it's an Arch packaging issue, let me know and I will post my question in the appropriate forum. Thanks.

**Comment by unknwon** [comment]
> [...], but I cannot find those in the list of files provided by the Archlinux package. Furthermore, I cannot find any of the CSS and image assets that are served by the Gogs web server in the package files.

Depending on which exact version you're running, most likely all default template files and static files are embedded into the binary so you won't find them on disk.

If you enable the trace level, Gogs tells you where the "custom" directory it is using:

```ini
[log]
LEVEL = Trace
```

e.g.
```
...
[TRACE] Custom path: [...]/gogs/gogs/custom
...
```

Then just put "templates" directory under there.

Source files of template files are in https://github.com/gogs/gogs/tree/main/templates and static files are in https://github.com/gogs/gogs/tree/main/public

Let me know if you have other questions!

**Comment by scossu** [comment]
Thanks @unknwon . 

This looks awkward:

```
2022/06/30 15:32:41 [TRACE] Work directory: /usr/bin
2022/06/30 15:32:41 [TRACE] Custom path: /usr/bin/custom
```

I don't see a setting in  `app.ini` to set that, is there one?

**Reply by unknwon** [reply]
You can't set the custom directory path through `app.ini` because it is used to find the `app.ini`. 😂

You may use env var `GOGS_CUSTOM` or `--config` CLI flag to specify the path for the custom directory. Once you changed the path, make sure to move over existing content in `/usr/bin/custom` to the new directory, so your current `app.ini` (and whatever others) could continue to be recognized by Gogs.

**Reply by scossu** [reply]
Thanks for the extra info. I will bring this up with the Arch community as it seems like a packaging issue. 

There is no `/usr/bin/custom` in my install.
